### PR TITLE
fixed formatting on to_csv() for categorical datetypes

### PR DESCRIPTION
**Pull request by kyleolaughlin**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/core/internals/blocks.py [hunk 1]
```diff
@@ -2050,19 +2050,23 @@ def to_native_types(
 ) -> np.ndarray:
     """ convert to our native types format """
     values = ensure_wrapped_if_datetimelike(values)
+    if isinstance(values, Categorical) and isinstance(values.categories[0], Timestamp):
```
Comment (mzeitlin11) [inline pandas/core/internals/blocks.py:2053]: You should be able to use `.dtype` to get the category type in a more stable way. Also, why is this timestamp specific? Do other datetimelike values work (and are tested?)

File: pandas/core/internals/blocks.py [hunk 2]
```diff
@@ -2050,19 +2050,23 @@ def to_native_types(
...
+        values = values.astype('datetime64[ns]')
+        values = DatetimeArray(values)
 
     if isinstance(values, (DatetimeArray, TimedeltaArray)):
         result = values._format_native_types(na_rep=na_rep, **kwargs)
         result = result.astype(object, copy=False)
         return result
 
+
```
Comment (mzeitlin11) [inline pandas/core/internals/blocks.py:2062]: Please remove the unrelated change here and in lines below

**Review by mzeitlin11 — COMMENTED** [review]
Thanks for the pr @kyleolaughlin. Some comments, plus looks like you accidentally committed the file `ctypes.__init__.cover`. 

Also, first thing any pr needs is tests!

**Comment by simonjayhawkins** [comment]
Thanks @kyleolaughlin for the PR. what is the issue number this PR is addressing?

**Comment by mzeitlin11** [comment]
Thanks for contributing @kyleolaughlin, closing as stale for now, but please ping to reopen if you'd like to continue working on this.
